core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -277,10 +277,30 @@ public void checkCompatibility(MappedFieldType other, List<String> conflicts) {
         if (!names().equals(other.names())) {
             conflicts.add("mapper [" + names().fullName() + "] has different index_name");
         }
-
         if (Objects.equals(similarity(), other.similarity()) == false) {
             conflicts.add("mapper [" + names().fullName() + "] has different similarity");
         }
+
+        if (strict) {
+            if (omitNorms() != other.omitNorms()) {
```

Comment:
We already have the check for norms going the "bad" direction above:
````
if (omitNorms() && !other.omitNorms()) {
    conflicts.add("mapper [" + names().fullName() + "] cannot enable norms (`norms.enabled`)");
}
```

The check in `strict` is for when norms change at all (even the good direction), since `strict` means no changes are allowed. Would it be better if I make it explicit this is the "ok" direction with the check?